src/distilabel/llm/litellm.py

```diff
@@ -0,0 +1,169 @@
+        api_version: Optional[str] = None,
+        api_key: Optional[str] = None,
+        model_list: Optional[list] = None,
+        mock_response: Optional[str] = None,
+        force_timeout: Optional[int] = 600,
+        custom_llm_provider: Optional[str] = None,
+    ) -> "GenerateOutput":
+        """Generates `num_generations` responses for the given input using the [LiteLLM async client](https://github.com/BerriAI/litellm).
+
+        Args:
+            input: a single input in chat format to generate responses for.
+            num_generations: the number of generations to create per input. Defaults to
+                `1`.
+            functions: a list of functions to apply to the conversation messages. Defaults to
+                `None`.
+            function_call: the name of the function to call within the conversation. Defaults
+                to `None`.
+            temperature: the temperature to use for the generation. Defaults to `1.0`.
+            top_p: the top-p value to use for the generation. Defaults to `1.0`.
+            stop: Up to 4 sequences where the LLM API will stop generating further tokens.
+                Defaults to `None`.
+            max_tokens: The maximum number of tokens in the generated completion. Defaults to
+                `None`.
+            presence_penalty: It is used to penalize new tokens based on their existence in the
+                text so far. Defaults to `None`.
+            frequency_penalty: It is used to penalize new tokens based on their frequency in the
+                text so far. Defaults to `None`.
+            logit_bias: Used to modify the probability of specific tokens appearing in the
+                completion. Defaults to `None`.
+            user: A unique identifier representing your end-user. This can help the LLM provider
+                to monitor and detect abuse. Defaults to `None`.
+            metadata: Pass in additional metadata to tag your completion calls - eg. prompt
+                version, details, etc. Defaults to `None`.
+            api_base: Base URL for the API. Defaults to `None`.
+            api_version: API version. Defaults to `None`.
+            api_key: API key. Defaults to `None`.
+            model_list: List of api base, version, keys. Defaults to `None`.
+            mock_response: If provided, return a mock completion response for testing or debugging
+                purposes. Defaults to `None`.
+            force_timeout: The maximum execution time in seconds for the completion request.
+                Defaults to `600`.
+            custom_llm_provider: Used for Non-OpenAI LLMs, Example usage for bedrock, set(iterable)
+                model="amazon.titan-tg1-large" and custom_llm_provider="bedrock". Defaults to
+                `None`.
+
+        Returns:
+            A list of lists of strings containing the generated responses for each input.
+        """
+
+        completion = await self._aclient(  # type: ignore
```

Comment:
Does it support `n`? Some do not as e.g. `Mistral`, so could you check that the generations are consistent when running some models other than the OpenAI ones? Thanks in advance!

We've seem some issues with the `n` paramater, and wanted to double check how is `LiteLLM` handling that

Reply:
Sure